### PR TITLE
moved labels to the same device as logits for opt, GPTJ , codeine and pix2struct models

### PR DESCRIPTION
**Pull request by sushmanthreddy**
# What does this PR do?

As suggested in the [#22561 ](https://github.com/huggingface/transformers/issues/22561) , moved labels to the same device as logits for OPT, GPTj, codegen and pix2struct models.

I am new to open source pls give me suggestions for keeping better pr.

### CODE REVIEW

**Comment by sushmanthreddy** [comment]
### there are some things to be changed i will keep pr after that changes

**Review by sgugger — APPROVED** [review]
Thanks for your contribution!
